### PR TITLE
KIN-299 File chunks implementation

### PR DESCRIPTION
**Pull request by Mauriciocr207**
[KIN-299](https://loopcrack.atlassian.net/browse/KIN-299) Small [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=loopcrack-org&utm_campaign=badge_info)

### Problem

It is needed a way to deal with big files uploaded by the users.

### Solution

Implement chunk files for bigger files uploaded by the users

### Requirements checklist

- [x] The routes must have a route where the filepond script can be able to send patch request
- [x] The CtrlApiFiles must manage the patch request where can be able to manage big files in chunks.
- [x] The FileManager class must have a method that merge all fragments of a file sended in chunks.

### Testing

- [x] Remember install all dependencies with `npm install` and run the project with `npm run dev:admin`
- [x] change `.env` variable  `CI_ENVIRONMENT = development` to `CI_ENVIRONMENT = production`
- [x] Into the filepond configuration setted from the class CtrlTestFiles you can see the values `chunkUploads = true` and `chunkSize = 100000` that means that now filepond can try to process files bigger than the `chunkSize` setted in bytes. 
- [x] Try to upload a file you know is bigger than `100000 bytes`, then you can see how the file is uploading. 

## Resources

[Reference Filepond Chunks](https://pqina.nl/filepond/docs/api/server/#process-chunks) 

## Demo

https://github.com/loopcrack-org/Kinub/assets/70463645/cf1eadb5-6d8b-48b7-b348-7eab09868c4f

[KIN-299]: https://loopcrack.atlassian.net/browse/KIN-299?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/Controllers/CtrlApiFiles.php [hunk 1]
```diff
@@ -83,6 +83,26 @@ public function processTemporalFile()
...
+            $offset     = $this->request->header('Upload-Offset')->getValue();
+
+            $folder  = FILES_TEMP_DIRECTORY . $key;
+            $fileTmp = "{$folder}/{$fileName}";
+
+            $temporalFileLength = FileManager::mergeChunckFiles($fileTmp, $fileData, $offset);
+
+            return $this->response->setStatusCode(204);
+        } catch (Throwable $th) {
+            return $this->response->setStatusCode(500, json_encode('Ha ocurrido un error mientras se cargaba el archivo'));
```
Comment (LUGAMAFE) [inline app/Controllers/CtrlApiFiles.php:102]:
```suggestion
             return $this->response->setStatusCode(Response::HTTP_NO_CONTENT);
        } catch (Throwable $th) {
            return $this->response->setStatusCode(Response::HTTP_INTERNAL_SERVER_ERROR , json_encode('Ha ocurrido un error mientras se cargaba el archivo'));
        }
```
Suggestion for better code readability and maintenance
@Mauriciocr207, I recommend using constant variables for HTTP status codes throughout this file. 
This approach not only enhances code readability but also makes the reasons behind each status code more clear.
Ensure to import the `use CodeIgniter\HTTP\Response;` class.

**Review by LUGAMAFE — CHANGES_REQUESTED** [review]
Same problem as @CheloChidoone 
image

**Comment by Mauriciocr207** [comment]
> @Mauriciocr207, We encountered an error, it shows us this error on console while uploading a big file and just stays processing it, and also it doesn't let us click to cancel it. Please check if you have any errors or let us know if we missed a step to test, we followed your test instructions so we don't think we missed anything. image
> 
> CC: @CarlosJCD

Sorry, I forgot to say that you need to set the project on production mode, if you don´t, the xmlHttpRequest includes the html of the codeigniter button, and the request fail.

Here the error:
Captura de pantalla (1445)

Here the correcto way to do it
image

**Review by LUGAMAFE — APPROVED** [review]
Hey @Mauriciocr207 is working now! Nice! But please make the changes about the http response codes in all the file begore merging please

**Review by CarlosMay7 — APPROVED** [review]

**Review by Oswaldo-Chan — APPROVED** [review]
Good job @Mauriciocr207!

**Review by CarlosJCD — APPROVED** [review]

**Review by CheloChidoone — APPROVED** [review]

**Pull request merged**